pennylane/qchem/tapering.py

```diff
@@ -485,7 +494,8 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
 
     # taper the HF observable using the symmetries obtained from the molecular hamiltonian
     fermop_taper = _taper_pauli_sentence(ferm_ps, generators, paulixops, paulix_sector)
-    fermop_mat = _binary_matrix(fermop_taper.ops, len(fermop_taper.wires))
+    fermop_ps = pauli_sentence(fermop_taper)
+    fermop_mat = _binary_matrix_from_pws(list(fermop_ps), len(fermop_taper.wires))
```

Comment:
Here are some new benchmarks, looks like we are no longer performing as much better as our previous estimates (more deviation for HF tapering).

|   Molecule   |  Ham Improvement |  HF Improvement |
|--------------|-----------------|-----------------|
|     H2       |     1.302x      |     0.998x      |
|     H4       |     0.975x      |     0.914x      |
|     H6       |     1.13x       |     0.589x      |
|     H8       |     1.904x      |     0.943x      |
|     H10      |     3.761x      |     1.109x      |